optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -947,4 +947,163 @@ default Constraint rewardConfigurableBigDecimal(String constraintName, Function<
+            ToIntFunction<A> matchWeigher);
+
+    /**
+     * Positively or negatively impact the {@link Score} by the constraintWeight multiplied by the match weight.
+     * Otherwise as defined by {@link #impact(String, Score)}.
+     * <p>
+     * Use {@code penalizeLong(...)} or {@code rewardLong(...)} unless you intend to mix positive and negative weights.
+     * @param constraintName never null, shows up in {@link ConstraintMatchTotal} during score justification
+     * @param constraintWeight never null
+     * @param matchWeigher never null, the result of this function (matchWeight) is multiplied by the constraintWeight
+     * @return never null
+     */
+    default Constraint impactLong(String constraintName, Score<?> constraintWeight, ToLongFunction<A> matchWeigher) {
+        return impactLong(getConstraintFactory().getDefaultConstraintPackage(), constraintName, constraintWeight,
+                matchWeigher);
+    }
+
+    /**
+     * As defined by {@link #impactLong(String, Score, ToLongFunction)}.
+     * @param constraintPackage never null
+     * @param constraintName never null
+     * @param constraintWeight never null
+     * @param matchWeigher never null
+     * @return never null
+     */
+    Constraint impactLong(String constraintPackage, String constraintName, Score<?> constraintWeight,
+            ToLongFunction<A> matchWeigher);
+
+    /**
+     * Positively or negatively impact the {@link Score} by the constraintWeight multiplied by the match weight.
+     * Otherwise as defined by {@link #impact(String, Score)}.
+     * <p>
+     * Use {@code penalizeBigDecimal(...)} or {@code rewardBigDecimal(...)} unless you intend to mix positive and
+     * negative weights.
+     * @param constraintName never null, shows up in {@link ConstraintMatchTotal} during score justification
+     * @param constraintWeight never null
+     * @param matchWeigher never null, the result of this function (matchWeight) is multiplied by the constraintWeight
+     * @return never null
+     */
+    default Constraint impactBigDecimal(String constraintName, Score<?> constraintWeight,
+            Function<A, BigDecimal> matchWeigher) {
+        return impactBigDecimal(getConstraintFactory().getDefaultConstraintPackage(), constraintName,
+                constraintWeight, matchWeigher);
+    }
+
+    /**
+     * As defined by {@link #impactBigDecimal(String, Score, Function)}.
+     * @param constraintPackage never null
+     * @param constraintName never null
+     * @param constraintWeight never null
+     * @param matchWeigher never null
+     * @return never null
+     */
+    Constraint impactBigDecimal(String constraintPackage, String constraintName, Score<?> constraintWeight,
+            Function<A, BigDecimal> matchWeigher);
+
+    /**
+     * Positively or negatively impact the {@link Score} by the {@link ConstraintWeight} multiplied by the match weight.
+     * Otherwise as defined by {@link #impactConfigurable(String)}.
```

Comment:
This link (and other links to `impactConfigurable` below) is broken; I haven't found such a method.

The same for Bi,Tri and QuadConstraintStream classes.

Reply:
Thanks, good catch.